crates/red_knot_module_resolver/src/resolver.rs

```diff
@@ -143,67 +155,300 @@ impl RawModuleResolutionSettings {
     /// This validation should probably be done outside of Salsa?
     ///
     /// [module resolution order]: https://typing.readthedocs.io/en/latest/spec/distributing.html#import-resolution-ordering
-    fn into_configuration_settings(self) -> ModuleResolutionSettings {
+    fn into_configuration_settings(
+        self,
+        current_directory: &SystemPath,
+    ) -> ModuleResolutionSettings {
         let RawModuleResolutionSettings {
             target_version,
             extra_paths,
             workspace_root,
-            site_packages,
+            site_packages: site_packages_setting,
             custom_typeshed,
         } = self;
 
-        let mut paths: Vec<ModuleResolutionPathBuf> = extra_paths
+        let mut static_search_paths: OrderSet<SearchPathRoot> = extra_paths
```

Comment:
I've realised that using an `OrderSet` here isn't sufficient to enforce that there'll be no duplication in the final list of search paths. `ModuleResolutionPathBuf::FirstParty(SystemPathBuf("/src"))` and `ModuleResolutionPathBuf::EditableInstall(SystemPathBuf("/src"))` will hash differently (due to them being different enum variants. But we shouldn't add the second one to the list of search paths for us to consider if the first one is already in the list, since they point to the same underlying path on disk :/

Reply:
> What about `/src` and `/src/foo`

That's bad coding practice (as Carl mentioned elsewhere), and will probably end up with surprising results for the user, but the runtime adds them both to `sys.path` in that situation, and we should aim to emulate the runtime in that regard. Possibly we could emit some kind of warning telling the user that this probably will have bad effects down the line, but at the same time there might be a reason for doing this (quite unusual) thing, so that might also be annoying.